src/components/workspace/BuildProject/BuildProject.tsx

```diff
@@ -638,9 +644,29 @@ const BuildProject: FC<Props> = ({ projectId, contract, updateContract }) => {
     if (activeProject?.network) {
       setEnvironment(activeProject.network);
     }
-    if (activeProject?.selectedContract) {
-      setSelectedContract(activeProject.selectedContract);
-      deployForm.setFieldsValue({ contract: activeProject.selectedContract });
+    const selectedABIPath = activeProject?.selectedContract;
+
+    if (selectedABIPath) {
+      const correspondingScriptPath = selectedABIPath.replace('.abi', '.ts');
```

Comment:
- These strings are repeated in other parts of code.
- The replace goes over the whole string, and `.abi` can be any part of it. For example, `.abi`, `.abi.foo.cpp` or `h.abibi.txt`.